cli/flags.rs

```diff
@@ -226,12 +232,32 @@ pub enum CheckFlag {
   Local,
 }
 
-impl Default for CheckFlag {
+impl Default for TypecheckMode {
   fn default() -> Self {
     Self::All
```

Comment:
> Discussed offline. `--check` should default to type checking only local files; to type check all files (including remote) we should do `--check=all`. Similar to how `--no-check` skips all, or `--no-check=remote` type checks only local files.

_Originally posted by @bartlomieju in https://github.com/denoland/deno/issues/11786#issuecomment-1077700276_

I think this should be done in this PR for 1.21. It's not a workflow-breaking change so it needs no transition period if we're doing it anyway. It's just an immediate improvement in usability.

Reply:
That's already implemented in this PR.

Reply:
But `deno check` aliases `deno cache` and doesn't seem to have an `--all` flag. Also the temporary case where `deno run main.ts` is called, `DENO_FUTURE_CHECK` is not set and the warning is shown, it will use `TypecheckMode::All`. I think this default needs to be changed.

Reply:
This is now implemented.